spring-cloud-gcp-starters/spring-cloud-gcp-starter-sql/src/main/java/org/springframework/cloud/gcp/sql/autoconfig/AppEngineCloudSqlJdbcInfoProvider.java

```diff
@@ -46,8 +46,7 @@ public String getJdbcDriverClass() {
 
 	@Override
 	public String getJdbcUrl() {
-		return String.format("jdbc:google:mysql://%s/%s?user=%s&password=%s",
-				this.properties.getInstanceConnectionName(), this.properties.getDatabaseName(),
-				this.properties.getUserName(), this.properties.getPassword());
+		return String.format("jdbc:google:mysql://%s/%s",
```

Comment:
Why removing the username and password?
If an instance has username and password set, don't we need to specify them here?

Reply:
That's main point of this change: everything rest is fully up to the existing `DataSourceProperties` and `DataSourceAutoConfiguration` in Spring Boot.

Isn't it @saturnism ?

Reply:
It sounds like we should be setting username and password in the `AppEngineCloudSqlJdbcInfoProvider` `@Bean` declaration in `GcpCloudSqlAutoConfiguration`.
Unless I'm missing something and the App Engine driver doesn't do anything with the SQL username and password.

Reply:
username/password is configured in the datasource, can be overwritten by spring.datasource.username and spring.datasource.password.  It does not seem necessary as part of URL construction.